sources/packages/backend/apps/api/src/services/application-restriction-bypass/application-restriction-bypass.service.ts

```diff
@@ -0,0 +1,49 @@
+import { Injectable } from "@nestjs/common";
+import {
+  ApplicationRestrictionBypass,
+  ApplicationStatus,
+  RecordDataModelService,
+} from "@sims/sims-db";
+import { ApplicationRestrictionBypassSummaryAPIOutDTO } from "apps/api/src/route-controllers/application-restriction-bypass/models/application-restriction-bypass.dto";
+import { DataSource } from "typeorm";
+
+/**
+ * Service layer for application restriction bypasses.
+ */
+@Injectable()
+export class ApplicationRestrictionBypassService extends RecordDataModelService<ApplicationRestrictionBypass> {
+  constructor(dataSource: DataSource) {
+    super(dataSource.getRepository(ApplicationRestrictionBypass));
+  }
+
+  /**
+   * Returns all restriction bypasses for a given application.
+   * @param applicationId id of the application to retrieve restriction bypasses.
+   * @returns application restriction bypasses.
+   */
+  async getApplicationRestrictionBypasses(
+    applicationId: number,
+  ): Promise<ApplicationRestrictionBypassSummaryAPIOutDTO[]> {
```

Comment:
Services should to return the DTOs. DTOs are exclusively meant to be used to encapsulate the data transferred over the network. So the service must return `ApplicationRestrictionBypass[]` as it is and transformation should happen at controller level transforming `ApplicationRestrictionBypass` to `ApplicationRestrictionBypassSummaryAPIOutDTO`

Reply:
Updated.